core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -203,6 +223,85 @@ oppia.factory('CollectionObjectFactory', [
       return skillList;
     };
 
+    // Gets a new ID for a skill. This should be of the same form as in the
+    // backend.
+    Collection.prototype.getNewSkillId = function() {
```

Comment:
You may want to mention in the documentation that this method has side effects (the skillIdCount/nextSkillId field is going to increment)